vxfreeswitch/voice.py

```diff
@@ -474,7 +476,7 @@ def send_outbound_message(self, client, message):
     def client_answered(self, client):
         """Function that is called when the ChannelAnswer event is received.
         Fires the deferred related to the outbound call"""
-        d = self._unanswered_channels.pop(client.get_address(), None)
+        d = self._unanswered_channels.get(client.get_address(), None)
```

Comment:
Yes, in `send_outbound_message` we used to yield on the `pop`, meaning that when we got to this point of the client answering, there wouldn't be anything in the dictionary, so we couldn't callback on the deferred.

So the first thing to do is to not pop in the `send_outbound_message`, and only pop here. But then you get an issue with when the client disconnects before the call is answered. In the `deregister_client`, we need access to unanswered channel to errback on it. But then you have this deferred stuck in the dictionary that won't ever get popped.

So I decided to not pop in both the `deregister_client` and the `client_answered` functions, and instead do a get in `send_outbound_message`, and after yielding on that deferred, then only pop it from the dictionary.